scipy/stats/morestats.py

```diff
@@ -1122,9 +1140,39 @@ def boxcox_normmax(x, brack=(-2.0, 2.0), method='pearsonr'):
 
     >>> plt.show()
 
+    Alternatively, we can define our own `optimizer` function.
+
+    >>> from scipy import optimize
+    >>> def optimizer(fun, args):
+    ...     return optimize.minimize_scalar(fun, bounds=(0, 1), args=args,
+    ...                                     method="bounded")
+    >>> stats.boxcox_normmax(x, optimizer=optimizer)
+    0.999...
     """
+    # If optimizer is not given, define default 'brent' optimizer.
+    if optimizer is None:
+
+        # Set default value for `brack`.
+        if brack is None:
+            brack = (-2.0, 2.0)
 
-    def _pearsonr(x, brack):
+        def _optimizer(func, args):
+            return optimize.brent(func, args=args, brack=brack)
+
+    # Otherwise check optimizer.
+    else:
+        if not callable(optimizer):
+            raise ValueError("`optimizer` must be a callable")
+
+        if brack is not None:
+            raise ValueError("`brack` must be None if `optimizer` is given")
+
+        # `optimizer` is expected to return a `OptimizeResult` object, we here get the
+        # solution to the optimization problem.
+        def _optimizer(func, args):
+            return optimizer(func, args).x.item()
```

Comment:
```suggestion
            return optimizer(func, args=args).x.item()
```
With this change `optimize=optimize.minimize_scalar` will work.